source/util/hex_float.h

```diff
@@ -662,12 +659,19 @@ inline uint8_t get_nibble_from_character(int character) {
   const char* dec = "0123456789";
   const char* lower = "abcdef";
   const char* upper = "ABCDEF";
-  const char* p = nullptr;
-  if ((p = strchr(dec, character))) {
+  const char* p = p = strchr(dec, character);
```

Comment:
Interestingly, this line seems to break the gcc build (see "checks have failed" below).